docs/sdg/sdg-flow-yaml.md

```diff
@@ -0,0 +1,164 @@
+
+The pipeline author needs some way of disambiguating between these multiple models - i.e. the definition of each `LLMBlock` needs to specify a particular model.
+
+Right now the `Pipeline` constructor takes two relevant parameters - the OpenAI client instance, and the model ID for the default model. It's important to note that this model ID is defined by the user at runtime, and it may not match the model IDs that the pipeline author used.
+
+The use cases will be:
+
+1. Most LLMBlock definitions will use the default teacher model - and we can make the semantic that if the pipeline author doesn't specify a model in an `LLMBlock`, the default in `PipelineContext.model_id` is used.
+2. In cases where a model+adapter is served, it should usually be the pipeline author defining both the model serving configuration and the model's use in an `LLMBlock`. So the pipeline author can name the model in the serving config and reference it by that name in the pipeline config.
+3. In cases where a model+adapter is served with a name that doesn't match that chosen by the pipeline author, the user can supply a mapping between the name used in the `LLMBlock` and the name used in the user's serving config.
+
+In future, it may make sense to define some model serving config (e.g. which adapter to use for SDG) as part of a flow definition. We will address this in a future if enhancement, if needed.
+
+### CLI Integration
+
+As of the current version of `ilab`, it supports `simple` and `full` as parameters to `--pipeline` to select one of the two types of built-in pipelines included in the library. Once we have support for loading custom pipelines, we need a way for these to be specified with the CLI. Here is a proposed CLI UX for this:
+
+> `ilab data generate`
+
+Use the default pipeline, `simple`.
+
+> `ilab data generate --pipeline full`
+
+Use the built-in `full` pipeline.
+
+> `ilab data generate --pipeline full --pipeline-extend path/to/config.yaml`
+
+Use the `full` pipeline, but extend it with the pipeline defined in the provided config file.
+
+> `ilab data generate --pipeline none --pipeline-extend path/to/other_config.yaml`
+
+Do not use a built-in pipeline. Use only the pipeline in the custom configuration.
+
+### Future CLI Improvements
+
+A possible improvement would be to have a well-defined place on the filesystem where custom pipeline configs can be automatically loaded and included as options to the `--pipeline` parameter.
+
+For example, if the config format included new parameters, `name: full-extended` and `extends: full`, and the CLI discovered and loaded it automatically, we could support `--pipeline full-extended` without needing the additional `--pipeline-extend` option.
```

Comment:
again I think this is out of the scope of this design doc but I feel this worth its own way to achieve "compose" (where chaining is just the simplest composition).

Reply:
Noted and responded to [here](https://github.com/instructlab/dev-docs/pull/109/files/f7663a515bb6535a536e43ee5839dc4913d1e96d#r1671729188)

And ... this is under "Future CLI Improvements". It's merely acknowledging ways the design might evolve in future, which is a useful thing to do in a design doc rather than simply reject as out of scope any ideas for the future from contributors to the discussion

Reply:
sounds good!